util/plugins/eu.esdihumboldt.util/src/eu/esdihumboldt/util/geometry/interpolation/ArcInterpolation.java

```diff
@@ -0,0 +1,253 @@
+
+		angle = getAngleBetweenTwoPoints(arcCoordinates[1], arcCoordinates[2], radius);
+
+		// Add coordinates up to next arc coordinates
+		generatedCoordinates.addAll(generateLineStringCoordinates(arcCoordinates[1],
+				arcCoordinates[2], center, radius, angle, NEXT_COORDINATE_DISTANCE));
+
+		// Add third Arc coordinate
+		generatedCoordinates.add(pointToGrid(arcCoordinates[2]));
+
+		// now, we have all coordinates of line string. So then just create it.
+		LineString lineString = null;
+		try {
+			lineString = new GeometryFactory()
+					.createLineString(generatedCoordinates.toArray(new Coordinate[0]));
+		} catch (Exception ex) {
+			log.error("Error in interpolation of arc", ex);
+		}
+
+		return lineString;
+	}
+
+	private double getAngleBetweenTwoPoints(Coordinate c1, Coordinate c2, double radius) {
+		double dx = c1.x - c2.x;
+		double dy = c1.y - c2.y;
+		double distance = Math.sqrt(dx * dx + dy * dy);
+
+		double angle = Math
+				.toDegrees(Math.acos(1 - (Math.pow(distance, 2) / (2 * Math.pow(radius, 2)))));
+
+		return angle;
+	}
+
+	private List<Coordinate> generateLineStringCoordinates(Coordinate current,
+			Coordinate nextArcCoordinate, Coordinate center, double radius, double maxAngle,
+			double nextPointDistance) {
+
+		List<Coordinate> tempList = new ArrayList<>();
+
+		double deservedNeighbourAngle = 0;
+		Coordinate deservedNeighour = null;
+
+		while (deservedNeighour == null) {
+			// all possible neighbor
+			List<Coordinate> allNeighbours = getNeighbouringCoordinates(current, nextPointDistance);
+
+			// will find deserved neighbor
+			for (Coordinate c : allNeighbours) {
+				double distance = Math
+						.sqrt(Math.pow((c.x - center.x), 2) + Math.pow((c.y - center.y), 2));
+
+				if (Math.abs(distance - radius) <= MAX_POSITIONAL_ERROR) {
+
+					deservedNeighbourAngle = getAngleBetweenTwoPoints(c, nextArcCoordinate, radius);
+
+					if (deservedNeighbourAngle < maxAngle) {
+						deservedNeighour = c;
+						break;
+					}
+				}
+			}
+			// half the next point distance
+			nextPointDistance /= 2;
+		}
+
+		// we found neighbor coordinate, we add nearest grid point of that arc
+		// point
+		tempList.add(pointToGrid(deservedNeighour));
+
+		// if we reached to the Arc coordinate then just return list, else call
+		// this method again
+		if (!isArcCoordinateNeighbour(deservedNeighour, nextArcCoordinate, MAX_POSITIONAL_ERROR)) {
+			tempList.addAll(generateLineStringCoordinates(deservedNeighour, nextArcCoordinate,
+					center, radius, deservedNeighbourAngle, NEXT_COORDINATE_DISTANCE));
+		}
+
+		return tempList;
+	}
+
+	private boolean isArcCoordinateNeighbour(Coordinate current, Coordinate nextArcCoordinate,
+			double e) {
+
+		double distance = Math.sqrt(Math.pow((current.x - nextArcCoordinate.x), 2)
+				+ Math.pow((current.y - nextArcCoordinate.y), 2));
+		if (distance <= e)
+			return true;
+
+		return false;
+	}
+
+	private List<Coordinate> getNeighbouringCoordinates(Coordinate current,
```

Comment:
This method gets all the neighbour coordinates of Arc not grid. First, algorithm search for the neighbour coordinates of arc then it moves most suited neighbour to grid point (nearest).